tt_metal/internal/program.hpp

```diff
@@ -0,0 +1,35 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <string_view>
+#include "types.hpp"
+
+#include "tt_metal/impl/kernels/kernel_types.hpp"
+#include "tt_metal/impl/buffers/circular_buffer_types.hpp"
+
+//==================================================
+//                  INTERNAL PROGRAM MANAGEMENT
+//==================================================
+
+namespace tt::tt_metal{
+namespace v1 {
+
+//==================================================
+//                 INTERNAL PROGRAM FUNCTIONS
+//==================================================
+
+/**
+ * @brief Launches a program on the device.
+ *
+ * @param device The device on which to launch the program.
+ * @param program The program to execute.
+ * @param wait_until_cores_done If true, waits until cores have completed execution.
+ * @param force_slow_dispatch If true, forces slow dispatch mode.
+ */
+void LaunchProgram(Device device, Program program, bool wait_until_cores_done = true, bool force_slow_dispatch = false);
```

Comment:
launch program options instead of default values?

Reply:
The second bool is getting removed on current PR soon to be merged.  https://github.com/tenstorrent/tt-metal/pull/13755. This won't cause a merge conflict. How do we plan to deal with these changes?

Reply:
We can remove it when the #13755 is merged.   When the implementation is in place we can pull this out.